### PR TITLE
fix: use a dummy report sender with no reporter URL

### PR DESCRIPTION
**Pull request by vladklokun**
## PR Type:
Bug fix

___
## PR Description:
This PR introduces a dummy report sender to prevent panic when trying to send events with no server URL provided. The dummy report sender stands in for a proper report sender but does not report anywhere. It is used when the EventReceiverURL in the configuration is empty. The PR also includes an update to the 'kubescape/backend' dependency.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`utils/reportsender.go`: Introduced a new struct 'dummyReportSender' with methods that mimic a real report sender but do not perform any actions. This is used as a placeholder when no server URL is provided.
`utils/typesutils.go`: Modified the 'NewSessionObj' function to use the dummy report sender when the EventReceiverURL in the configuration is empty.
`go.mod`: Updated the 'kubescape/backend' dependency from v0.0.10 to v0.0.11.
`go.sum`: Updated the checksums related to the 'kubescape/backend' dependency.
</details>

___
## Overview
This PR fixes a panic when trying to send events with no server URL provided.

**[Signed Commits](../CONTRIBUTING.md#sign-off-per-commit)**
- [x] Yes, I signed my commits.

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: utils/reportsender.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"time"
+)
+
+type dummyReportSender struct{}
```
Comment (dwertent) [inline utils/reportsender.go:7]: This code should be in the `backend` package

**Review by dwertent — CHANGES_REQUESTED** [review]

File: utils/reportsender.go (hunk 1 above)
Comment (vladklokun) [reply utils/reportsender.go:7]: https://github.com/kubescape/backend/pull/14

Comment (vladklokun) [reply utils/reportsender.go]: We merged the PR.

**Comment by vladklokun** [comment]
@dwertent 
We should also create a release for this fix, right? I added the label.

**Review by dwertent — APPROVED** [review]

**Pull request merged**